pkg/database/vercode.go

```diff
@@ -195,6 +195,29 @@ func (db *Database) FindVerificationCodeByUUID(uuid string) (*VerificationCode,
 	return &vc, nil
 }
 
+// ExpireCode saves a verification code as expired.
+func (db *Database) ExpireCode(uuid string) (*VerificationCode, error) {
+	var vc VerificationCode
+	err := db.db.Transaction(func(tx *gorm.DB) error {
+		if err := tx.
```

Comment:
Looking at this again, why do we attempt to find the record? Why not just:

```go
now := time.Now().UTC()
db.Table("verification_codes").Where("uuid = ?").Update(VerificationCode{
  ExpiresAt: now,
  LongExpiresAt: now,
}).Error
```

That avoids the need to take out a lock.

Reply:
See smaller PRs does work. I'm used to spanner where blind-writes are disallowed.

Nice!